src/main/java/com/ashkan/moviesapi/services/implementations/MovieRentalServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+
+    @Autowired
+    public MovieRentalServiceImpl(MovieRentalRepository movieRentalRepository,
+                                  MovieRepository movieRepository,
+                                  MemberRepository memberRepository,
+                                  MovieCatalogRepository movieCatalogRepository) {
+        this.movieRentalRepository = movieRentalRepository;
+        this.movieRepository = movieRepository;
+        this.memberRepository = memberRepository;
+        this.movieCatalogRepository = movieCatalogRepository;
+    }
+
+    @Override
+    public List<MovieRental> findAll() {
+        return movieRentalRepository.findAll();
+    }
+
+    @Override
+    public MovieRental findById(int id) {
+        return movieRentalRepository.findById(id)
+                .orElseThrow(() -> new MovieRentalNotFoundException(id));
+    }
+
+    @Override
+    public void save(MovieRental movieRental) {
+        Movie movie = movieRepository.findById(movieRental.getMovieId())
+                .orElseThrow(() -> new MovieRentalNotFoundException(movieRental.getMovieId()));
+
+        Member member = memberRepository.findById(movieRental.getMemberId())
+                .orElseThrow(() -> new MovieRentalNotFoundException(movieRental.getMemberId()));
+
+        MovieCatalog movieCatalog = movieCatalogRepository.findById(movieRental.getMovieId())
+                .orElseThrow(() -> new RuntimeException("Movie catalog not found " + movieRental.getMovieId()));
+
+        Timestamp currentDate = new Timestamp(System.currentTimeMillis());
+        Timestamp returnDate = new Timestamp(System.currentTimeMillis() + constants.DAYS_TO_ADD);
+
+        movieRental.setMovieByMovieId(movie);
+        movieRental.setMemberByMemberId(member);
+        movieRental.setDate(currentDate);
+        movieRental.setToReturnDate(returnDate);
+        movieRental.setStatus("Rented");
+
+        if (movieCatalog.getCopies() > 0) {
+            movieCatalog.setCopies(movieCatalog.getCopies() - 1);
+        } else {
+            throw new MovieRentalBadRequestException_NoCopies();
+        }
+
+        movieRentalRepository.save(movieRental);
+    }
+
+    @Override
+    public void deleteById(int id) {
+        movieRentalRepository.deleteById(id);
+    }
+
+    @Override
+    public void patchStatus(MovieRental movieRental, int id) {
+        MovieRental movieRentalFind = movieRentalRepository.findById(id)
+                .orElseThrow(() -> new MovieRentalNotFoundException(id));
+
+        MovieCatalog movieCatalogFind = movieCatalogRepository.findById(movieRentalFind.getMovieId())
+                .orElseThrow(() -> new RuntimeException("Movie Catalog not found" + movieRentalFind.getMovieId()));
+
+        if (movieRentalFind.getStatus() == null) {
+            throw new RuntimeException("This movie needs to be rented before been returned!");
+        }
+
+        if (movieRental.getStatus().equals("Returned")) {
```

Comment:
Please use the equalsIgnoreCase function in order to accept the value **_returned_** as well.

Reply:
done.